pimd/pim_str.h

```diff
@@ -26,6 +26,8 @@
 
 #include <prefix.h>
 
+#define PREFIX_SG prefix_sg
```

Comment:
Doesn't it look weird? Do we use such a pattern for structs at all?

Reply:
Our motive is to use the macro PREFIX_SG for both PIMV4 and PIMV6 , since both the daemons need to consume the same code.

Reply:
union is not going to work in such a case?

Reply:
Different types is correct, but the way to select the correct with a macro like this isn't.

(Making it an union is not helpful, we know at compile time whether we're doing IPv4 or IPv6 and we should use that to our advantage.)